pkg/evetestkit/utils.go

```diff
@@ -0,0 +1,487 @@
+package evetestkit
+
+import (
+	"fmt"
+	"io"
+	"net"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/docker/docker/pkg/namesgenerator"
+	"github.com/dustin/go-humanize"
+	"github.com/lf-edge/eden/pkg/controller"
+	"github.com/lf-edge/eden/pkg/defaults"
+	"github.com/lf-edge/eden/pkg/device"
+	"github.com/lf-edge/eden/pkg/eve"
+	"github.com/lf-edge/eden/pkg/openevec"
+	"github.com/lf-edge/eden/pkg/projects"
+	"github.com/lf-edge/eden/pkg/tests"
+	"github.com/lf-edge/eden/pkg/utils"
+	"github.com/tmc/scp"
+	"golang.org/x/crypto/ssh"
+	"golang.org/x/exp/rand"
+)
+
+const (
+	// DefaultSSHPass is a default ssh password for the VM running on the EVE node
+	DefaultSSHPass = "passw0rd"
+	// DefaultSSHUser is a default ssh user for the VM running on the EVE node
+	DefaultSSHUser = "ubuntu"
+	// DefaultCloudConfig is a default cloud-init configuration for the VM which just
+	// enables ssh password authentication and sets the password to "passw0rd".
+	DefaultCloudConfig = "#cloud-config\npassword: " + DefaultSSHPass + "\nchpasswd: { expire: False }\nssh_pwauth: True\n"
+)
+
+// appInstanceConfig is a struct that holds the information about the app
+// running on the EVE node
+type appInstanceConfig struct {
+	name    string
+	sshPort string
+	sshUser string
+	sshPass string
+}
+
+// EveNode is a struct that holds the information about the remote node
+type EveNode struct {
+	controller *openevec.OpenEVEC
+	edgenode   *device.Ctx
+	cfg        *openevec.EdenSetupArgs
+	tc         *projects.TestContext
+	apps       []appInstanceConfig
+	ip         string
+}
+
+type VMOption func(*EveNode)
+
+var (
+	controllerVerbosiry = "warn"
+	edenConfEnv         = defaults.DefaultConfigEnv
+)
+
+func getEdenConfig() (*openevec.EdenSetupArgs, error) {
+	conf := os.Getenv(edenConfEnv)
+	configName := utils.GetConfig(conf)
+	cfg, err := openevec.FromViper(configName, controllerVerbosiry)
+	if err != nil {
+		return nil, fmt.Errorf("can't get the config: %w", err)
+	}
+
+	return cfg, nil
+}
+
+func getOpenEVEC() (*openevec.OpenEVEC, *openevec.EdenSetupArgs, error) {
+	cfg, err := getEdenConfig()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return openevec.CreateOpenEVEC(cfg), cfg, nil
+}
+
+// CreateEveNode creates a new EveNode struct
+func CreateEveNode(node *device.Ctx, tc *projects.TestContext) (*EveNode, error) {
```

Comment:
Should we keep this exported even though EveNode is returned by `InitilizeTest`?